### ISSUE TITLE
Proposal for the removal of medium steel

### ISSUE THREAD

**Issue by XygenSS** [CleverRaven/Cataclysm-DDA#76540]
### Is your feature request related to a problem? Please describe.

The current steel situation gives us 6 grades of steel to work with:
`steel`, `mild steel`, `medium steel`, `high steel`, `hardened steel`, `tempered steel`.
`steel` is eventually going to be obsoleted, which gives us 5 grades to stay.
as far as performance as a weapon or armor goes the hierarchy is tempered > hardened > high > medium > mild.

Mild steel is extremely abundant and is the easiest to work with, and (as of now) can easily be case-hardened.
Medium steel is hardly ever found anywhere except in crafting and on traders, and can be tempered.
High steel is also difficult to find except in crafting and on traders, but cannot be hardened or tempered.

The problem is that high steel is in an awkward position.
1. Since you must craft your own graded steel most of the time, there is little reason not to go straight to hardened steel (which is better than high steel and easily made from mild steel) or tempered steel (the best there is).
2. Since tempered steel is repaired using medium steel chunks there's very little reason to ever craft high steel at all. Both medium and high carbon steel can be tempered IRL but only medium steel can be transformed to tempered steel ingame. 
3. Where exactly you would find medium vs high carbon steel ingame is fuzzy and sometimes arbitrary. Both grades of steel are often used in the same application IRL and it depends on the manufacturer and the specific application. Different industries categorize steel grades differently and not all sources provide the exact carbon content. 

### Solution you would like.

I propose the removal of medium steel, which leaves us with:

1. Mild steel, which represents all commonly-found low-carbon steel (0.1~0.2% C). Ubiquitous, easy to work with.
2. Case hardened steel which is (relatively easily) made from mild steel. The case hardening process also could be revisited; it is possible to harden steel using an acetylene torch and then quenching in oil.
3. High carbon steel, anything above 0.5% C. The intent is to combine medium and high steel into one. It is found on heavy-duty and high-strength applications such as hand tools, blades, automotive springs, axles, and gears, wear-resistant parts, wire ropes, music wire, etc.
4. Tempered steel, the best there is, made from high carbon steel.

Skill requirements for crafting should be adjusted. Working with high steel should require +1 skill level, but tempering it should only add +1 additional instead of +2. 

We are already mostly ignoring the existence of high alloy content steels, most importantly stainless steel. For the purposes of the average survivor I don't think we need to fuss over whether our steel is 0.4% carbon or 0.8% carbon.

### Describe alternatives you have considered.

Another way of differentiating steel grades is repairability. Mild steel is easily welded. High carbon steel has poor weldability but is possible to weld. Case hardened steel could be made unrepairable; alternatively it could revert to mild steel after a major repair. Tempered steel can be left as-is until we have a more detailed heat-treatment model.

**Comment by XygenSS** [comment]
Paging @Drew4484, I hope you don't mind

**Comment by Drew4484** [comment]
I can't speak to how other contributors have distributed the various grades but the intent was to represent a (very narrow) selection of what would be found IRL, rather than do a linear **worst -> best** line.

The skill jump for qt_steel was intentional. It represents a demanding technique with a high failure rate.

Alloy steels would be basically impossible to replicate for a survivor, so unless you want to add hundreds of flavors that would have to be melted down and separated before the survivor could do anything with them, we can skip that. Long term durability isn't something we currently simulate, so (for now), stainless/rust is a non-issue, though I have considered chromium oxide coatings.

**Comment by XygenSS** [comment]
> The skill jump for qt_steel was intentional. It represents a demanding technique with a high failure rate.

We can have multiple tiers of tempering proficiencies and bump their failure chance modifiers until satisfactory.
Also, if high steel gets a +1 skill bump, 6mm tempered armor recipes would require skill level 10, that's why I suggest lowering the tempering skill req. by 1.

> represent a (very narrow) selection of what would be found IRL, rather than do a linear worst -> best line.

I don't think we have enough differing characteristics for the carbon steel grades not to be a worst-best line. Currently the differences are weapon damage values, armor protection values and skill requirement. That is pretty "linear."

> Alloy steels

They don't need to be *replicated*, since the survivor can simply salvage them. Neither do they need to be implemented, in my opinion - even something like HSLA steel for vehicle frames wouldn't necessarily translate to a better garage-forged sword. Like you've said we don't have long term weathering and rusting of steel parts. They're just alloyed mild steel, so we simply treat them as mild steel (#75850).

**Comment by Drew4484** [comment]
Grades of proficiencies are pretty tedious, especially when the fail chance is already high and the resource cost would get expensive fast. Difficulty 10 crafts effectively do not exist in the core game, a difficulty 10 crafts are typically beyond the capacity of a survivor with whatever is left of the technology and production base. 

You're looking at the material qualities in reverse. We have recipes for armor made from carpet. We have recipes for making armor out of 2x4s. We have armor that is deliberately designed to be sub-optimal. There are lots of bad weapons, useless ammo, etc.  High carbon steel isn't the _best_ option, but players may, for whatever reason, want to make it. In a perfect world (with paid developers to implement it), we'd have materials as a selectable option for various crafts, instead, we have some bad items to make.

Salvaging alloys gets into a whole separate set of problems. Different alloys aren't going to weld together properly, some can only handle incompatible heat ranges from others, some do or don't work with resistance welding, some need non-standard rods or shielding gas, some have so much non-iron and non-carbon that you're not _really_ working with 'steel' anymore. So in game what you'd get is a poor-quality amalgam, which we represent with budget_steel. Unless you're specifically salvaging the exact same parts off the exact same car (and might need to even be from the same date range of manufacture), you'll wind up with similar but functionally different materials. 

I don't argue that our metalworking is an accurate representation of real world metalworking. It cuts corners to fit into the game, without requiring the player to know as much about steel as (at minimum) I do. So there's definitely compromises, I think the ones we have made are good enough without requiring things like selectable materials. But if you want to PR some changes, I 100% support you doing so, because I'm not the final word on any part of CDDA.

**Comment by ADekema** [comment]
I think that the easiest part of this to solve is simply the availability of medium and high steel. Just find out what things in real life are made from medium and high steel and make sure that those same things in game are made from the same steel.

**Comment by XygenSS** [comment]
> I think that the easiest part of this to solve is simply the availability of medium and high steel. Just find out what things in real life are made from medium and high steel and make sure that those same things in game are made from the same steel.

Many things in real life are sometimes made of medium steel and sometimes made of high steel. Depending on the industry and manufacturer their definition of medium or high steel may be different as well.

**Comment by Drew4484** [comment]
> I think that the easiest part of this to solve is simply the availability of medium and high steel. Just find out what things in real life are made from medium and high steel and make sure that those same things in game are made from the same steel.

That's the simplest solution. Unfortunately it would be a little bit of work.
![image](https://github.com/user-attachments/assets/dfea1aa7-779f-498e-8b7b-d34c8d2cf3f9)

**Comment by ADekema** [comment]
> That's the simplest solution. Unfortunately it would be a little bit of work.

We don't need to do every single thing. Just a few specific items like landsmower blades, vehicle frames and maybe something like axeheads would be enough. Just stuff that a player could reasonably find that could give a usefull amound of medium or high steel.

**Comment by CoroNaut** [comment]
I would be for removing medium steel if steel is also going eventually. Then we could have a simple system:
mild steel -> hardened steel
high steel -> tempered steel

This is simple and effectively gets the point across that they can be slightly upgraded without introducing too much different steels. For a player that probably knows nothing about steel and steel accessories besides in CDDA, this works.

As for skill requirement, making the tempered steel easier to craft would not be the best IMO. If tempered is better than hardened, then it should be a bit harder, or at the very least, finding high steel should be harder.

**Comment by Hyperseeker** [comment]
> Difficulty 10 crafts effectively do not exist in the core game, a difficulty 10 crafts are typically beyond the capacity of a survivor with whatever is left of the technology and production base.

Even though it's certainly possible, mechanically speaking? Is there any documentation to substantiate this line of thinking? `GAME_BALANCE.md` has nothing on it.

**Comment by harakka** [comment]
> I would be for removing medium steel if steel is also going eventually. Then we could have a simple system: mild steel -> hardened steel high steel -> tempered steel
> 
> This is simple and effectively gets the point across that they can be slightly upgraded without introducing too much different steels. For a player that probably knows nothing about steel and steel accessories besides in CDDA, this works.

This isn't how steel works so whatever the player knowledge, it's not going to be the solution. We want to represent how this stuff works irl, even at lowered level, we're not designing generic videogame style item progression.

**Comment by CoroNaut** [comment]
I dumbed it down quite a bit as to not re-iterate the complexities and what was said in the original issue. But from what I gathered from the original issue, this is at pretty much what the stated solution is. If steel and medium steel are removed, then yeah, these four are all that's left. I was pointing out that it was simple to understand for players who wouldn't actually have this knowledge IRL. Navigating through tons of crafting recipes for tons of varying steels would be a bit cumbersome. If instead this issue doesn't end up going through and we get more steels, that would be fine too. Its just that it may be less intuitive.

**Comment by XygenSS** [comment]
> This isn't how steel works so whatever the player knowledge, it's not going to be the solution. We want to represent how this stuff works irl, even at lowered level, we're not designing generic videogame style item progression.

Which part of it is not congruent to reality? The current system is wrong as well, you should be able to temper high steel, and they should be harder to work with instead of all base steel grades being 5 skill level difficulty.

**Comment by fairyarmadillo** [comment]
According to google, medium steel is frequently used in construction for beams, joints, joiner plates. It's also what the tanks in many boilers and water heaters are made of, and also appears in things like axle and crankshafts and rail tracks, wheels, and other train parts.

It would be pretty simple to make a construction site for a steel-frame building with some stacks of medium steel beams lying around. These could also be placed at the steel mills that already exist. It sounds like some engine parts could be medium steel, and the game has tons of water heaters and tanks. So if more of the stuff is needed in the world, that'd be an easy place to start.

**Comment by Drew4484** [comment]
> > This isn't how steel works so whatever the player knowledge, it's not going to be the solution. We want to represent how this stuff works irl, even at lowered level, we're not designing generic videogame style item progression.
> 
> Which part of it is not congruent to reality? The current system is wrong as well, you should be able to temper high steel, and they should be harder to work with instead of all base steel grades being 5 skill level difficulty.

The harder the steel, the more difficult it is to work. You certainly _can_ temper higher carbon steels, but the result (if done correctly) is going to approach/surpass the tools a survivor has access to. The current qt_steel was based on a grade that would remain workable by the equipment in the game. So if we want stronger/tougher/harder, we'd need better tools, the sort that can't be found in a hobbyist's garage.

**Comment by Drew4484** [comment]
> According to google, medium steel is frequently used in construction for beams, joints, joiner plates. It's also what the tanks in many boilers and water heaters are made of, and also appears in things like axle and crankshafts and rail tracks, wheels, and other train parts.
> 
> It would be pretty simple to make a construction site for a steel-frame building with some stacks of medium steel beams lying around. These could also be placed at the steel mills that already exist. It sounds like some engine parts could be medium steel, and the game has tons of water heaters and tanks. So if more of the stuff is needed in the world, that'd be an easy place to start.

I started working on this #71835 with the goal of adding factory-produced vehicle frames with slightly higher durability (attempting to represent modern automotive steels) that would break down into a mix of grades when disassembled. Unfortunately this was a little overly ambitious a month after our second kid was born so it's just been gathering dust. I plan to someday finish it up but if you want to salvage from my PR or do your own thing in that space, I'll happily give whatever support I can.

**Comment by IdleSol** [comment]
Go down to the subway and smash the rails. You will have tons of medium steel.
https://cdda-guide.nornagon.net/item/railroad_track_item

Or find an NPC who sells steel. He has all kinds. Easily swaps normal steel for the necessary steel.

My personal opinion. The problem is not in the number of different types of steel. The problem is that:
1. All recipes are in the same book
2. Require the same level
3. Serve for the same items (armor and weapons)
4. Roughly equal in terms of time (+1 day to make a better version means nothing).

As a consequence, it makes no sense to craft something not from the best version.

**Comment by fairyarmadillo** [comment]
> > According to google, medium steel is frequently used in construction for beams, joints, joiner plates. It's also what the tanks in many boilers and water heaters are made of, and also appears in things like axle and crankshafts and rail tracks, wheels, and other train parts.
> > It would be pretty simple to make a construction site for a steel-frame building with some stacks of medium steel beams lying around. These could also be placed at the steel mills that already exist. It sounds like some engine parts could be medium steel, and the game has tons of water heaters and tanks. So if more of the stuff is needed in the world, that'd be an easy place to start.
> 
> I started working on this #71835 with the goal of adding factory-produced vehicle frames with slightly higher durability (attempting to represent modern automotive steels) that would break down into a mix of grades when disassembled. Unfortunately this was a little overly ambitious a month after our second kid was born so it's just been gathering dust. I plan to someday finish it up but if you want to salvage from my PR or do your own thing in that space, I'll happily give whatever support I can.

I converted sheet metal over to be LCS a while back, but stopped there because so many recipes (ie makeshift hammer) rely on generic steel, and I didn't want to mess up early game progression.

I'm not submitting PRs these days, but I think anyone reading this who is might want to start by updating those tools.

**Comment by IdleSol** [comment]
I don't know about that. A quick google search says that tempered steel is the further processing of hardened steel.

How realistic would it be if all tempered steel items required a hardened steel item? And a hardened steel item would require a regular steel item?

Or what it might look like using the hammer as an example. (Because it has a steel part and a non-steel part.)

Prerequisite:
`high steel` = `medium steel` + `coal` + `time` = `mild steel` + `coal` x2 + `time`
`medium steel` = `mild steel` + `coal` + `time`
`mild steel` = `medium steel`+ `time` = `high steel` + `time`

And we remove the steel grade from the name. Leaving just the steel object. This leaves one version of the item (weapons and armor). And the material requirements rely on meaning. Where it doesn't matter is mild steel. Armor and weapons medium or high steel. Crafted tools: medium. Industrial tools: high. And so on.

Then the process of making a hammer would look like this:

1. Make the wooden handle of the hammer
2. Make a steel hammer head.
3. Choosing:
a. Completing the fabrication of the steel hammer
b. Harden the steel hammer head to produce a hardened hammer head
4. Choosing:
a. Completing the fabrication of the hardened hammer
b. Tempering the hardened hammer head to produce a tempered hammer head
5. Completing the fabrication of the tempered hammer

**Comment by Drew4484** [comment]
> > I started working on this #71835 with the goal of adding factory-produced vehicle frames with slightly higher durability (attempting to represent modern automotive steels) that would break down into a mix of grades when disassembled. Unfortunately this was a little overly ambitious a month after our second kid was born so it's just been gathering dust. I plan to someday finish it up but if you want to salvage from my PR or do your own thing in that space, I'll happily give whatever support I can.
> 
> I converted sheet metal over to be LCS a while back, but stopped there because so many recipes (ie makeshift hammer) rely on generic steel, and I didn't want to mess up early game progression.
> 
> I'm not submitting PRs these days, but I think anyone reading this who is might want to start by updating those tools.
> 
> > As a consequence, it makes no sense to craft something not from the best version.
> 
> I've seen players echoing this a lot, and being frustrated about tempered steel's craft times while not realizing they should be starting with lower grades, pretty much for this reason.

The biggest issue here is not that the craft times are too long, but that a majority of our crafts are/have been far too short. Of course the heavy tempered plate suit stands out, it's an incredibly complicated masterpiece. But the old plate armor could be made in an afternoon. A lot of the oldest content has not been rigorously reviewed for things like crafting time, but should.

**Comment by Drew4484** [comment]
> I don't know about that. A quick google search says that tempered steel is the further processing of hardened steel.
> 
> How realistic would it be if all tempered steel items required a hardened steel item? And a hardened steel item would require a regular steel item?
> 
> Or what it might look like using the hammer as an example. (Because it has a steel part and a non-steel part.)
> 
> Prerequisite: `high steel` = `medium steel` + `coal` + `time` = `mild steel` + `coal` x2 + `time` `medium steel` = `mild steel` + `coal` + `time` `mild steel` = `medium steel`+ `time` = `high steel` + `time`
> 
> And we remove the steel grade from the name. Leaving just the steel object. This leaves one version of the item (weapons and armor). And the material requirements rely on meaning. Where it doesn't matter is mild steel. Armor and weapons medium or high steel. Crafted tools: medium. Industrial tools: high. And so on.
> 
> Then the process of making a hammer would look like this:
> 
> 1. Make the wooden handle of the hammer
> 2. Make a steel hammer head.
> 3. Choosing:
>    a. Completing the fabrication of the steel hammer
>    b. Hardening the steel hammer head to produce a hardened hammer head
> 4. Choosing:
>    a. Completing the fabrication of the hardened hammer
>    b. Tempering the hardened hammer head to produce a tempered hammer head
> 5. Completing the fabrication of the tempered hammer

There's something like 4k different alloys that we call 'steel'. Each one has unique properties. The same temperature that is needed to temper one steel might ruin the alloy of another. If you don't know what kind of steel you have (which is difficult to identify outside of a lab!), tempering may destroy the item you're working on. 

We _can_ add craft stages, that sort of happens when the suit is assembled from the individually crafted torso/arms/legs. I'm all for breaking our longer/difficult claims down into smaller steps, but it is a lot of work.

Having multiple items with the same name is a no-go.

**Comment by IdleSol** [comment]
>Having multiple items with the same name is a no-go.

I'm not suggesting making three versions with the same name. I'm suggesting to keep only one version and rename it “steel”. But this steel version, can be made from three different types of steel. With different resource and time costs.

But that seems to be beyond the scope of this thread. Personally my opinion, one of two options should be pursued:

1. Reduce the number of types of steel
2. Reduce the number of “same” items. Even if it contradicts reality

**Comment by Drew4484** [comment]
> > Having multiple items with the same name is a no-go.
> 
> I'm not suggesting making three versions with the same name. I'm suggesting to keep only one version and rename it “steel”. But this steel version, can be made from three different types of steel. With different resource and time costs.
> 
> But that seems to be beyond the scope of this thread. Personally my opinion, one of two options should be pursued:
> 
> 1. Reduce the number of types of steel
> 2. Reduce the number of “same” items. Even if it contradicts reality

The initial graded plate had five grades of steel for three suits of armor. 15 recipes. Add XL and XS versions, now we're at 45. Add alternate options like creating each individual piece and then combine it, 90. 

What you're proposing (if I understand your post correctly) is three suits of armor. Three recipes. Plus XL/XS, 9. Now add recipes for every combination of steel grade that can be accepted. Of the 130 chunks of X steel required for a full heavy suit, how are you going to break that down to allow different materials? If the time is dependent on the amounts of each material used, you'll need a separate recipe for every possible combination to account for the crafting times.

So some millions of recipes later, you could accept every possible combination of steel chunks with the appropriate crafting times based on inputs. 

Or, alternatively, add C++ to allow dynamic material selection, which is probably a quick and easy project that none of the developers has gotten around to in the last decade or so.

**Comment by IdleSol** [comment]
```
--    "result": "armor_lc_brigandine",
++    "result": "armor_STELL_brigandine",
    "type": "recipe",
...
    "using": [
...
--      [ "lc_steel_standard", 5 ],
++      [ "steel_standard_example_MILD", 5 ],
...
```
We're removing the following recipes:
-    "result": "armor_mc_brigandine",
-    "result": "armor_hc_brigandine",

```
  {
    "id": "steel_standard_example_MILD",
    "type": "requirement",
    "components": [
      [ [ "lc_steel_lump", 1 ], [ "lc_steel_chunk", 4 ], [ "coal", 0], ["time_multiplier", 1], ["time_add", 0] ],
      [ [ "mc_steel_lump", 1 ], [ "mc_steel_chunk", 4 ], [ "coal", 0], ["time_multiplier", 1], ["time_add", 120 ] ],
      [ [ "hc_steel_lump", 1 ], [ "hc_steel_chunk", 4 ], [ "coal", 0], ["time_multiplier", 1], ["time_add", 180 ] ],
    ]
  },
  {
    "id": "steel_standard_example_MEDIUM",
    "type": "requirement",
    "components": [
      [ [ "lc_steel_lump", 1 ], [ "lc_steel_chunk", 4 ], [ "coal", 8] ["time_multiplier", 1], ["time_add", 120] ],
      [ [ "mc_steel_lump", 1 ], [ "mc_steel_chunk", 4 ], [ "coal", 0], ["time_multiplier", 1], ["time_add", 0 ] ],
      [ [ "hc_steel_lump", 1 ], [ "hc_steel_chunk", 4 ], [ "coal", 0], ["time_multiplier", 1], ["time_add", 180 ] ],
    ]
  },
  {
    "id": "steel_standard_example_HIGH",
    "type": "requirement",
    "components": [
      [ [ "lc_steel_lump", 1 ], [ "lc_steel_chunk", 4 ], [ "coal", 8], ["time_multiplier", 1], ["time_add", 180] ],
      [ [ "mc_steel_lump", 1 ], [ "mc_steel_chunk", 4 ], [ "coal", 8], ["time_multiplier", 1], ["time_add", 60 ] ],
      [ [ "hc_steel_lump", 1 ], [ "hc_steel_chunk", 4 ], [ "coal", 28], ["time_multiplier", 1], ["time_add", 0 ] ],
    ]
  },
```
Explanation.

1. This is a rough draft.
2. I'm assuming that some items, should use different grades of steel. Conditionally. A steel rapier should only be made of `high steel`. And armor, only from `mild steel`. So I keep `steel_standard_example_XXX`. The recipe uses whichever variation makes the best sense. But the result will still be “steel item”

In the example above, I assume that the brigantine should be made of low steel and only low steel. It could then be subjected to hardening and tempering.

It might be more correct to have it made from a different steel. Then instead of `steel_standard_example_MILD` we use `steel_standard_example_MEDIUM` or `steel_standard_example_HIGH`. But only one of these three options. The result will always be a `steel brigantine`.

This can be simplified by reducing `steel_standard_example_XXX` to `steel_standard`. Which describes all types of steel and the rules for their interchangeability. Simpler, but less precise.

3. In the components section, a time component is added. Both the option of adding a fixed amount and adding a multiplier. I'm in favor of fixed time. But it's not important right now.
4. It is necessary to teach the game to perceive the new time component. I don't see any difficulty in doing this. But I don't know how to do it either, so I won't claim it's easy.

As a result, there won't be 100500 different recipes.

There will be 1 basic recipe. + 2 recipes for XL, XS. And additionally x2...x4 for different types, for example making by parts or remaking from ready-made ones.

P.S. Translated with DeepL

**Comment by IdleSol** [comment]
Alternative. Modify the “using” section.
```
--    "result": "armor_lc_brigandine",
++    "result": "armor_STELL_brigandine",
    "type": "recipe",
...
    "using": [
...
--      [ "lc_steel_standard", 5 ],
++      [
++        { [ "lc_steel_standard", 5 ], ["coal", 0], ["time_add", 0] },
++        { [ "mc_steel_standard", 5 ], ["coal", 8], ["time_add", 120] },
++        { [ "hc_steel_standard", 5 ], ["coal", 28], ["time_add", 180] }
++      ],
...
```
... AND [ `[ "lc_steel_standard", 5 ], ["coal", 0], ["time_add", 0] }` OR `{ [ "mc_steel_standard", 5 ], ["coal", 8], ["time_add", 120] }` OR `[ "hc_steel_standard", 5 ], ["coal", 28], ["time_add", 180]` ] AND ...

**Comment by Drew4484** [comment]
> ```
> --    "result": "armor_lc_brigandine",
> ++    "result": "armor_STELL_brigandine",
>     "type": "recipe",
> ...
>     "using": [
> ...
> --      [ "lc_steel_standard", 5 ],
> ++      [ "steel_standard_example_MILD", 5 ],
> ...
> ```
> 
> We're removing the following recipes:
> 
> * "result": "armor_mc_brigandine",
> * "result": "armor_hc_brigandine",
> 
> ```
>   {
>     "id": "steel_standard_example_MILD",
>     "type": "requirement",
>     "components": [
>       [ [ "lc_steel_lump", 1 ], [ "lc_steel_chunk", 4 ], [ "coal", 0], ["time_multiplier", 1], ["time_add", 0] ],
>       [ [ "mc_steel_lump", 1 ], [ "mc_steel_chunk", 4 ], [ "coal", 0], ["time_multiplier", 1], ["time_add", 120 ] ],
>       [ [ "hc_steel_lump", 1 ], [ "hc_steel_chunk", 4 ], [ "coal", 0], ["time_multiplier", 1], ["time_add", 180 ] ]
>     ]
>   },
>   {
>     "id": "steel_standard_example_MEDIUM",
>     "type": "requirement",
>     "components": [
>       [ [ "lc_steel_lump", 1 ], [ "lc_steel_chunk", 4 ], [ "coal", 8] ["time_multiplier", 1], ["time_add", 120] ],
>       [ [ "mc_steel_lump", 1 ], [ "mc_steel_chunk", 4 ], [ "coal", 0], ["time_multiplier", 1], ["time_add", 0 ] ]
>     ]
>   },
>   {
>     "id": "steel_standard_example_HIGH",
>     "type": "requirement",
>     "components": [
>       [ [ "lc_steel_lump", 1 ], [ "lc_steel_chunk", 4 ], [ "coal", 8], ["time_multiplier", 1], ["time_add", 180] ],
>       [ [ "mc_steel_lump", 1 ], [ "mc_steel_chunk", 4 ], [ "coal",28], ["time_multiplier", 1], ["time_add", 60 ] ],
>       [ [ "hc_steel_lump", 1 ], [ "hc_steel_chunk", 4 ], [ "coal", 0], ["time_multiplier", 1], ["time_add", 0 ] ]
>     ]
>   },
> ```
> 
> Explanation.
> 
> 1. This is a rough draft.
> 2. I'm assuming that some items, should use different grades of steel. Conditionally. A steel rapier should only be made of `high steel`. And armor, only from `mild steel`. So I keep `steel_standard_example_XXX`. The recipe uses whichever variation makes the best sense. But the result will still be “steel item”
> 
> In the example above, I assume that the brigantine should be made of `low steel` and only `low steel`. It could then be subjected to hardening and tempering.
> 
> It might be more correct to have it made from a different steel. Then instead of `steel_standard_example_MILD` we use `steel_standard_example_MEDIUM` or `steel_standard_example_HIGH`. But only one of these three options. The result will always be a `steel brigantine`.
> 
> This can be simplified by reducing `steel_standard_example_XXX` to `steel_standard`. Which describes all types of steel and the rules for their interchangeability. Simpler, but less precise.
> 
> 3. In the components section, a time component is added. Both the option of adding a fixed amount and adding a multiplier. I'm in favor of fixed time. But it's not important right now.
> 4. It is necessary to teach the game to perceive the new time component. I don't see any difficulty in doing this. But I don't know how to do it either, so I won't claim it's easy.
> 
> As a result, there won't be 100500 different recipes.
> 
> There will be 1 basic recipe. + 2 recipes for XL, XS. And additionally x2...x4 for different types, for example making by parts or remaking from ready-made ones.
> 
> P.S. Translated with DeepL

That would require rebuilding huge portions of the crafting code. That in and of itself isn't a bad thing, and if you want to submit a PR or a separate issue to discuss changes to crafting times/accepted materials, go for it. But you've gotten very far from arguing that medium steel should be removed from the game. I'm going to close this issue now, and I suggest you learn more about the current crafting system before you make more proposals related to it.
